### DISCUSSION TITLE
recommended template/project structure for multi-routes app

### DISCUSSION THREAD

**Discussion by bobanovo** [zauberzeug/nicegui#858] [Q&A] [answered]
### Question

What is the right structure (recommended) of nicegui app with multiple routes etc. ? Is there any boilerplate or template for this? Thanks a lot, Radek

**Comment by rodja** [comment]
Have you seen the [@ui.page decorator](https://nicegui.io/documentation#page) to define routes? Or are you looking for something like [APIRouter](https://fastapi.tiangolo.com/tutorial/bigger-applications/?h=apirouter#apirouter) for pages? What is your requirement?

**Reply by bobanovo** [reply]
Thanks a lot for your answer. Yes, I know this decorator, and use them, but if I have more routes, then one src file is almost unreadable. Therefore I need to divide it into more files for better maintenance.
**APIRouter** could be a way for this, thanks! I've read already about it, but didn't use it yet.

It would be good if any template exists for a more complex app, with more routes, sessions, etc. ;-) as a recommendation for better code with nicegui :-) ...

**Reply by bobanovo** [reply]
Hello, one additional question - I am trying to use APIRouter approach, but I have an issue with right response model. I understand APIRouter for classis API, when I am able to manage the own response, but in case nicegui, the response for @page decorater (in router is @router) is different, for API it is returning value/object/ but how do it for nicegui? ... Thanks a lot.

**Reply by rodja** [reply, marked as answer]
I've just created pull-request #897 which introduces a specalization of FastAPI's APIRouter to support pages. Could you review it and tell us if it works for your scenario?

**Reply by rodja** [reply]
It's now released 🎉
